sympy/matrices/determinant.py

```diff
@@ -828,3 +828,43 @@ def _minor_submatrix(M, i, j):
     cols = [a for a in range(M.cols) if a != j]
 
     return M.extract(rows, cols)
+
+
+def _permanent(M):
+    """Returns the permanent of a matrix as defined in [1]
+
+
+    Examples
+    ========
+
+    >>> from sympy import Matrix
+    >>> M = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    >>> M.permanent()
+    450
+
+    References
+    ==========
+
+    1. Prof. Frank Ben's notes: https://math.berkeley.edu/~bernd/ban275.pdf
```

Comment:
```suggestion
    .. [1] Prof. Frank Ben's notes: https://math.berkeley.edu/~bernd/ban275.pdf
```

Reply:
Also include a link to the Wikipedia article.